apps/dashboard/src/@/constants/client.ts

```diff
@@ -3,27 +3,25 @@ import {
   DASHBOARD_THIRDWEB_SECRET_KEY,
   IPFS_GATEWAY_URL,
 } from "@/constants/env";
+import { useMemo } from "react";
 import { createThirdwebClient } from "thirdweb";
 
-export const thirdwebClient = createThirdwebClient(
-  DASHBOARD_THIRDWEB_SECRET_KEY
-    ? {
-        secretKey: DASHBOARD_THIRDWEB_SECRET_KEY,
-        config: {
-          storage: {
-            gatewayUrl: IPFS_GATEWAY_URL,
-          },
-        },
-      }
-    : {
-        clientId: DASHBOARD_THIRDWEB_CLIENT_ID,
-        config: {
-          storage: {
-            gatewayUrl: IPFS_GATEWAY_URL,
-          },
-        },
+// returns a thirdweb client with optional JWT passed in
+export function getThirdwebClient(jwt?: string) {
+  return createThirdwebClient({
+    secretKey: jwt ? jwt : DASHBOARD_THIRDWEB_SECRET_KEY,
+    clientId: DASHBOARD_THIRDWEB_CLIENT_ID,
+    config: {
+      storage: {
+        gatewayUrl: IPFS_GATEWAY_URL,
       },
-);
+    },
+  });
+}
+
+export function useThirdwebClient(jwt?: string) {
```

Comment:
is the pattern now to use this everywhere in client land and the get fn above in ssr land?

Reply:
pretty much yes